app/Nova/Resources/Auth0Client.php

```diff
@@ -67,6 +71,19 @@ public function fields(NovaRequest $request): array
                     Auth0Tenant::Testing->value => Auth0Tenant::Testing->name,
                     Auth0Tenant::Production->value => Auth0Tenant::Production->name,
                 ]),
+            Text::make('Visible for integrator', static function (Auth0ClientModel $model) {
+                $auth0Client = $model->toDomain();
+                /** @var Integration $integration */
+                $integration = App::get(IntegrationRepository::class)->getById($auth0Client->integrationId);
+                $isVisible = $auth0Client->tenant !== Auth0Tenant::Acceptance &&
```

Comment:
I think you could add an extra method `Integration::isKeyVisibleForEnvironment`. It requires only a conversion from a specific environment to the environment inside the Integration namespace. But all those enums have the same value.

Reply:
Wouldn't you also need to take the `keyVisibility` into account?

Reply:
But maybe I can get the visibilty from environment.

Reply:
But the visibility is taken from the integration I think, see `$integration->getKeyVisibility()`

Reply:
I have implemented it, but I do not think it's ideal.